src/back/dispatcher.rs

```diff
@@ -81,6 +81,25 @@ impl DispatcherBuilder {
         self
     }
 
+    pub fn with_backends(mut self, mut new_backends: HashMap<ProviderID, BackEndHandler>) -> Self {
+        match &mut self.backends {
+            Some(backends) => {
+                for (provider_id, backend_handler) in new_backends.drain() {
+                    backends.insert(provider_id, backend_handler);
+                }
+            }
+            None => {
+                let mut map = HashMap::new();
+                for (provider_id, backend_handler) in new_backends.drain() {
+                    map.insert(provider_id, backend_handler);
+                }
+                self.backends = Some(map);
```

Comment:
Could it just be:
```rust
self.backends = Some(new_backends);
```
?

Reply:
Maybe there is some way to call `extend` on an `Option` to handle both cases

Reply:
See update